CMSgov/mint-app
# [Easi 4447] Nullable audit meta data

# EASI-4447

## Changes and Description

-  for audit records with meta data about records that can be deleted make this data nullable
    - Solution Subtask
    - CR
    - TDL
    - Collaborator
- Add integration test to verify that insert, update, and delete all function as expected, even if the record has been deleted.


## How to test this change
1. Run Unit tests

1. Create all these records
2. Update them
3. Delete them
4. Translate audits
5. Verify that the audits all continue to translate as expected

Verify all cases are handled
1. These changes specifically handle meta data. Please think about the other possible scenarios that we could run into. 
2. For example, are there any foreign key fields that point to data that can be deleted that this doesn't cover? Do we error if they can't be translated and should be? (for example plan document)
    a. Note, currently a document field will error if it can't be fetched from the database. Further edge cases of this can be addressed in a future PR if needed.



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
